### PR TITLE
Throw warning when reading NRRD05 files to upgrade to ITK >= 5.2

### LINKED ISSUE #1451: ITKReader does not read NRRD05 images with 'measurement frame' field
**Describe the bug**
ITKReader throws an error when reading NRRD05 images with the "measurement frame" field. Here's part of the traceback:
```
~/workspace/envs/sono-env/lib/python3.6/site-packages/monai/data/image_reader.py in _get_meta_dict(self, img)
    220             if key.startswith("ITK_original_"):
    221                 continue
--> 222             meta_dict[key] = img_meta_dict[key]
    223         meta_dict["origin"] = np.asarray(img.GetOrigin())
    224         meta_dict["spacing"] = np.asarray(img.GetSpacing())

~/workspace/envs/sono-env/lib/python3.6/site-packages/itk/ITKCommonBasePython.py in __getitem__(self, key)
    887         import itk
    888         obj = self.Get(key)
--> 889         return itk.down_cast(obj).GetMetaDataObjectValue()
    890     def __len__(self):
    891         return self.GetKeys().size()

~/workspace/envs/sono-env/lib/python3.6/site-packages/itkExtras.py in down_cast(obj)
   1198         raise RuntimeError(
   1199             "Can't downcast to a specialization of %s" %
-> 1200             className)
   1201     else:
   1202         return t.cast(obj)

RuntimeError: Can't downcast to a specialization of MetaDataObject
```
**To Reproduce**
Run
```
import numpy as np
import SimpleITK as sitk
from monai.data import ITKReader
from monai.transforms import LoadImage

fn = "tmp.seg.nrrd"
data = np.zeros((3,3,3))
im = sitk.GetImageFromArray(data)
im.SetMetaData("NRRD_measurement frame", "(1,0,0) (0,1,0) (0,0,1)")
sitk.WriteImage(im, fn)

LoadImage(reader=ITKReader())(fn)
```
Commenting out the `SetMetaData` line runs fine.

**Expected behavior**
ITKReader reads NRRD005 files.

**Environment**
```
================================
Printing MONAI config...
================================
MONAI version: 0.4.0
Numpy version: 1.17.4
Pytorch version: 1.6.0.dev20200528+cu101
MONAI flags: HAS_EXT = False, USE_COMPILED = False
MONAI rev id: 0563a4467fa602feca92d91c7f47261868d171a1

Optional dependencies:
Pytorch Ignite version: 0.4.2
Nibabel version: 3.2.1
scikit-image version: 0.17.2
Pillow version: 6.2.1
Tensorboard version: 2.4.0
gdown version: 3.12.2
TorchVision version: 0.7.0.dev20200528+cu101
ITK version: 5.1.2
tqdm version: 4.54.1
lmdb version: NOT INSTALLED or UNKNOWN VERSION.
psutil version: NOT INSTALLED or UNKNOWN VERSION.
```

**Additional context**
Is ignoring the `measurement frame` field like this here an acceptable fix? https://github.com/adamaji/MONAI/commit/13cbe07e5af795f67560189b36a1486d4ce69f0e If so, I can issue the PR.

### PR DESCRIPTION
**Pull request by adamaji**
Signed-off-by: Adam Aji <3487395+adamaji@users.noreply.github.com>

Fixes #1451 .

### Description
Reading NRRD05 files errors out when reading the "measurement frame" field, which was addressed in  https://github.com/InsightSoftwareConsortium/ITK/pull/2270 thanks to @thewtex . Since the change isn't in a stable build yet, here we ignore the problem field so that the image can still be read, and throw a warning pointing users to upgrade to the release candidate version (5.2rc2) for correct loading if desired.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).

### CODE REVIEW

**Review by wyli — APPROVED** [review]
thanks!

File: monai/data/image_reader.py [hunk 1]
```diff
@@ -219,6 +220,16 @@ def _get_meta_dict(self, img) -> Dict:
             # ignore deprecated, legacy members that cause issues
             if key.startswith("ITK_original_"):
                 continue
+            if (
+                key == "NRRD_measurement frame"
+                and int(itk.Version.GetITKMajorVersion()) == 5
+                and int(itk.Version.GetITKMinorVersion()) < 2
```
Comment (rijobro) [inline monai/data/image_reader.py:226]: This wouldn't work for ITK v4. Would need to be more like:

```python
if key == "NRRD" and (vMaj < 5 or vMin < 2)
```

Comment (adamaji) [reply monai/data/image_reader.py:226]: I think `((vMaj < 5) or (vMaj == 5 or vMin < 2))` right?

But does MONAI support ITK < 5? Should this section change too?
https://github.com/Project-MONAI/MONAI/blob/master/monai/data/image_reader.py#L128-L132

Comment (wyli) [reply monai/data/image_reader.py:226]: perhaps we should have a min. verison at the project level, `itk>=5.0`? @Nic-Ma @hjmjohnson we don't have thorough tests for all the versions, but it seems the demos are built with itk 5.x

Comment (rijobro) [reply monai/data/image_reader.py:226]: I think @aylward and @thewtex would like the minimum version to be set to `itk==5.2` once that's released. In the meantime, setting it to `itk==5.0` is fine.

**Comment by aylward** [comment]
LGTM!

---
Sent via phone.

**Review by Nic-Ma — APPROVED** [review]
Looks good to me.
I agree with @wyli to set the min ITK version as 5.0 for current status.

Thanks.

**Pull request merged**